File: firstnonrepeatingchar/FirstNonRepeatingCharacter.py
def firstNonRepeatingCharacter(string):
    dict = {}
    for str in string:
        try:
            frequency = dict[str]
            dict[str] = frequency + 1
        except:
            dict[str] = 1

    for key, value in dict.items():
        if value == 1:
            return key

    return -1


if __name__ == '__main__':
    string = "abcdcaf"
    print(firstNonRepeatingCharacter(string))

File: firstnonrepeatingchar/FirstNonRepeatingCharacter.java
public class FirstNonRepeatingCharacter {

    public static int firstNonRepeatingCharacter(String string) {

        char[] chars = string.toCharArray();
        int count = 0;
        boolean aux;
        while (chars.length > count) {
            aux = false;
            for (int i = 0; i < chars.length; i++) {
                char current = chars[count];
                char comparable = chars[i];
                if (current == comparable && i != count) {
                    aux = true;
                }
            }
            if (aux == false){
                return count;
            }
            count++;
        }
        return -1;
    }

    public static void main(String[] args) {
        System.out.println(firstNonRepeatingCharacter("abcdcaf"));
    }

}
